assets/scripts/streets/StreetNameCanvas.jsx

```diff
@@ -0,0 +1,86 @@
+import _ from 'lodash'
+import React from 'react'
+import StreetName from './StreetName'
+import StreetMetaData from './StreetMetaData'
+import { getStreet } from './data_model'
+import ReactDOM from 'react-dom'
+
+export default class StreetNameCanvas extends React.Component {
+  constructor (props) {
+    super(props)
+
+    var street = getStreet()
+    this.state = {
+      classNames: [],
+      street: street
+    }
+    this.streetUpdated = this.streetUpdated.bind(this)
+    this.updatePositions = this.updatePositions.bind(this)
+  }
+
+  componentDidMount () {
+    window.addEventListener('stmx:set_street', this.streetUpdated)
+    window.addEventListener('stmx:width_updated', this.streetUpdated)
+    window.addEventListener('resize', this.updatePositions)
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('stmx:set_street', this.streetUpdated)
+    window.removeEventListener('stmx:width_updated', this.streetUpdated)
+    window.removeEventListener('resize', this.updatePositions)
+  }
+
+  componentDidUpdate () {
+    this.updatePositions()
+  }
+
+  streetUpdated (e) {
+    const street = getStreet()
+    this.setState({street})
+  }
+
+  updatePositions () {
+    // TODO don't use the DOM when the whole app is react
+    const menuRect = document.querySelector('.menu-bar-right').getBoundingClientRect()
+
+    // TODO there might be a better way to do this than findDomNode
```

Comment:
Events make sense if you're communicating between two components that don't really have a relationship. When `StreetName` is a child of this component it seems the best way to communicate to a parent component is via a callback function passed as a prop. I guess we don't really need to pass the ref at all? `StreetName` could pass its dimensions through the callback function instead because that's all `StreetNameCanvas` needs to know